source/_posts/2015-04-08-so-you-want-an-ssl-certificate.markdown

```diff
@@ -0,0 +1,250 @@
+
+For RapidSSL you will also need more intermediate CAs, currently obtained from [here](https://knowledge.rapidssl.com/support/ssl-certificate-support/index?page=content&id=AR1548). You can get them as a bundle (we save it as *intermediate.crt*) or separately (we save these as *intermediate-primary.crt* and *intermediate-secondary.crt*).
+
+## Installing Certificates
+
+### Heroku
+
+You need the [SSL endpoint addon](https://devcenter.heroku.com/articles/ssl-endpoint) to upload the certificate to Heroku. The certificate must be a concatenation of your certificate and all intermediaries. For our RapidSSL certificates this is usually made as follows.
+
+
+```
+$ cat example.com.crt intermediate.crt > concatenated.crt
+
+$ heroku certs:add concatenated.crt example.com.key --app example-app
+
+Adding SSL endpoint to example-app... done
+example-app now served by tokyo-7171.herokussl.com
+Certificate details:
+    subject: /serialNumber=xyz/OU=GT65834870/OU=See www.rapidssl.com/resources/cps (c)13/OU=Domain Control Validated - RapidSSL(R)/CN=*.example.com
+    start date: 2013-01-13 18:21 EST
+    expire date: 2014-01-17 01:55 EST
+    common name(s): *.example.com, example.com
+    issuer: /serialNumber=yz/OU=GT65834870/OU=See www.rapidssl.com/resources/cps (c)13/OU=Domain Control Validated - RapidSSL(R)/CN=*.example.com
+    SSL certificate is verified by a root authority.
+```
+
+If you have a certificate already installed, update it.
+
+```
+$ heroku certs:update concatenated.crt example.com.key --app example-app
+```
+
+Heroku is known to patch common certificate issues during this process, such as certificates listed out of order.
+
+### OpsWorks
+
+OpsWorks offers 3 SSL fields in the application configuration (choose *Apps*, then *Edit* the app):
+
+<img src='/images/2015-04-08-so-you-want-an-ssl-certificate/opsworks.png'>
+
+The *SSL Certificate* should be the same contents of the *concatenated.crt* we made above.
+
+The *SSL Certificate key* should be the contents of *example.com.key*.
+
+The *SSL Certificates of Certification Authorities* should be left empty.
+
+### AWS ELB
+
+Login to AWS, go to *EC2 Instances*, click on *Load Balancers*, find your load-balancer and choose the *Listeners* tab. You can upload or choose a certificate here.
+
+<img src='/images/2015-04-08-so-you-want-an-ssl-certificate/elb.png'>
+
+The *Private Key* should be *example.com.key*.
+
+The *Public Key Certificate* should be the contents of *example.com.crt*.
+
+The *Certificate Chain* is the most problematic, it needs to list the entire chain, in our case it's a concatenation of *example.com.intermediate.crt*, then *intermediate-primary.crt*, and the root certificate for issuer of the latter, which is the Equifax Secure Certificate Authority from [here](https://www.geotrust.com/resources/root-certificates/).
```

Comment:
I honestly still don't totally understand this ordering thing for ELB, and why it seems to be so troublesome :(